LanguageFeatures/Augmentation-libraries/augmenting_declarations_A01_t01.dart

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if:
+/// - An augmenting declaration has no corresponding original declaration to
+///   apply to.
+///
+/// @description Checks that it is a compile-time error if an augmenting
+/// declaration has no corresponding original declaration to apply to
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment class C {}
```

Comment:
If the language team (1) accepts the proposal that I hinted at in https://github.com/dart-lang/language/issues/3647, or (2) tells me that we already have such a rule then this would be an error simply because it isn't in an augmentation library.

So we ought to have a test where it does occur in an augmentation library, and still there is no original declaration to augment.

Reply:
(This might be covered by tests further down, I haven't looked at them all).

Reply:
It is now OK to have an augmenting declaration in a library, so I'll close this thread.